lib/src/default_emoji_picker_view.dart

```diff
@@ -163,105 +215,77 @@ class _DefaultEmojiPickerViewState extends State<DefaultEmojiPickerView>
-          };
-
-          final onLongPressed = () {
-            if (!emoji.hasSkinTone || !widget.config.enableSkinTones) {
-              _closeSkinToneDialog();
-              return;
-            }
-            _closeSkinToneDialog();
-            _openSkinToneDialog(emoji, emojiSize, categoryEmoji, index);
-          };
-
-          return _buildButtonWidget(
-            onPressed: onPressed,
-            onLongPressed: onLongPressed,
-            child: _buildEmoji(
-              emojiSize,
-              categoryEmoji,
-              emoji,
-              widget.config.enableSkinTones,
-            ),
-          );
-        }).toList(),
-      ),
+          scrollDirection: Axis.vertical,
+          controller: _scrollController,
+          primary: false,
+          padding: widget.config.gridPadding,
+          crossAxisCount: widget.config.columns,
+          mainAxisSpacing: widget.config.verticalSpacing,
+          crossAxisSpacing: widget.config.horizontalSpacing,
+          children: [
+            for (int i = 0; i < categoryEmoji.emoji.length; i++)
+              _buildEmojiCell(
+                  emojiSize, categoryEmoji, categoryEmoji.emoji[i], i)
+          ]),
     );
   }
 
   /// Build and display Emoji centered of its parent
   Widget _buildEmoji(
     double emojiSize,
-    CategoryEmoji categoryEmoji,
     Emoji emoji,
     bool showSkinToneIndicator,
   ) {
-    // FittedBox needed for display, font scale settings
-    return FittedBox(
-      fit: BoxFit.fill,
-      child: Stack(children: [
-        emoji.hasSkinTone && showSkinToneIndicator
-            ? Positioned(
-                bottom: 0,
-                right: 0,
-                child: CustomPaint(
-                  size: const Size(8, 8),
-                  painter: TriangleShape(widget.config.skinToneIndicatorColor),
-                ),
-              )
-            : Container(),
-        Text(
-          emoji.emoji,
-          textScaleFactor: 1.0,
-          style: TextStyle(
-            fontSize: emojiSize,
-            backgroundColor: Colors.transparent,
-          ),
-        ),
-      ]),
+    final style = TextStyle(
+      fontSize: emojiSize,
+      backgroundColor: Colors.transparent,
+    );
+    final emojiText = Text(
+      emoji.emoji,
+      textScaleFactor: 1.0,
+      style: widget.config.emojiTextStyle == null
+          ? style
+          : widget.config.emojiTextStyle!.merge(style),
+    );
+
+    return Center(
+      child: emoji.hasSkinTone && showSkinToneIndicator
+          ? Container(
+              decoration: TriangleDecoration(
+                  color: widget.config.skinToneIndicatorColor, size: 8.0),
+              child: emojiText,
+            )
+          : emojiText,
     );
   }
 
   /// Build different Button based on ButtonMode
   Widget _buildButtonWidget({
     required VoidCallback onPressed,
-    required VoidCallback onLongPressed,
+    VoidCallback? onLongPressed,
```

Comment:
I don't see any situation this can become `null`, why removing `required` then?

Reply:
It is null for skin colors emojis. Used to be an empty handler, which is meh, in my opinion.